Fix /admin

alisonrclarke opened the pull request:
I've now got the admin interface working locally, with the settings as in **config_local.local.py**.

I've added some suggested [Talisman](https://github.com/GoogleCloudPlatform/flask-talisman) settings for the Kubernetes pods to **config.py** but they may need adjusting - @benjamin-bergia would you mind taking a look?

**Deployment instructions**

1. Before deploying: remove `APP_ENABLE_SECURE_HEADERS = False` from hepdata.cfg
2. After deploying, run the following command via `psql`:
    ```
    insert into alembic_version values ('2f63be7b7572');
    ```
3. Run `hepdata alembic upgrade` to apply the latest [upgrade](https://github.com/inveniosoftware/invenio-access/blob/master/invenio_access/alembic/04480be1593e_add_actionsystemroles.py) for `invenio_access`.

Fixes #224

alisonrclarke committed:
Revert "Use APP_DEFAULT_SECURE_HEADERS for local builds and tests"

b48160971001043977fb2ae568abb66a71b9aa9d

alisonrclarke committed:
Turn off most security options in local config files.

alisonrclarke committed:
Reinstate APP_ENABLE_SECURE_HEADERS=False for tests

alisonrclarke committed:
Re-add invenio-access to requirements and update invenio-* to latest 1.0 versions

invenio-access is needed to allow access to the admin interface

alisonrclarke committed:
Add talisman settings to main config file.

We'll need to remove `APP_ENABLE_SECURE_HEADERS = False` from hepdata.cfg

GraemeWatt reviewed (approved):
This seems to work on my local instance.  I was using a recent DB created under Python 3 without `invenio-access` installed, so the `access_*` tables were missing.  I had to insert `67ba0de65fbb` into the `alembic_version` table before running `hepdata alembic upgrade` to create the `access_*` tables, then add a user with `admin-access` from http://localhost:5000/admin/actionusers/ .  By default, all users have admin access unless an entry is added to the `access_actionsusers` table of the DB.  But the production DB already has the `access_*` tables and entries in the `access_actionsusers` table, so I think your instructions should work.  I'll test them on the QA DB before deploying in production.

I was wondering if there was a way to avoid duplication of the key/value pairs between `config.py` and `config_local.local.py` or `config_local.docker_compose.py`, since only three values are different, but I couldn't work out how to do it given the way that the variables of `config_local.py` are imported into `config.py`.

alisonrclarke committed:
Use the same talisman settings in main config.py as local version.

As https is done by ingress rather than the web pods, we don't enforce
https on the pods.

alisonrclarke commented:
Following conversations with Ben I've now used the previous local settings in the main config file.

If the access settings aren't correct when this is pushed to QA, you can use the [cli commands](https://invenio-accounts.readthedocs.io/en/latest/usage.html#command-line-interface) to add admin users.

alisonrclarke committed:
Merge branch 'master' into fix/admin

alisonrclarke committed:
Reinstate docs about APP_ENABLE_SECURE_HEADERS in .travis.yml

GraemeWatt committed:
INSTALL.rst: remove mention of 'APP_ENABLE_SECURE_HEADERS = False'
